### ISSUE TITLE
Feature request: Add drop down box to select a year when selecting a route.

### ISSUE THREAD

**Issue by espinosaluis** [espinosaluis/TrackMeViewer#44]
Something similar to what we have on the Android client.

Right now, all your routes show up in the drop down list. If you have many, it gets very hard to
find the one that you are interested. Actually I don't know what kind of sorting it has. I think we should have from newest to the oldest.

It would be nice to have an additional drop down box on top of the existing one where you can select the year first. If possible, only years that contain a route should show up. So if the user only saved routes in 2013 and 2014, only those two numbers should be there.

**Comment by xZise** [comment]
At the moment it should sort them by the ID descending which is only incrementing (so the order will be always from newest to oldest).

I'm not sure how allowing to selecting a year will help as this might still be overwhelming. Maybe a textual search is better.

**Comment by espinosaluis** [comment]
I don't think it would be overwhelming. I have been using TrackMe since 2007 and I use that feature all the time on my phone and it's great. I like to see the routes I have done every year. By the way, even if the sorting is by ID it's useless because when you export a route from the phone, they don't always get sorted by ID. It should be sorted by date at least on the viewer.

**Comment by xZise** [comment]
You need to specify what "sorted by date" means. It is sorted by start date of the trips because it assigns a new ID when you create a new trip. And that ID is only incrementing and thus the new trip will be sorted before the other trips.

**Comment by espinosaluis** [comment]
Yes, but you can export a route that you did 5 years ago and it will show up on top. That doesn't make sense.

**Comment by espinosaluis** [comment]
So as it is right now, the date is not used at all.

**Comment by xZise** [comment]
Ah okay importing might cause a previous trip to have a higher ID.
